src/extension.ts

```diff
@@ -1,10 +1,12 @@
-import { Prisma } from '@prisma/client/extension.js'
+import { Prisma, PrismaClient } from '@prisma/client/extension.js'
 
-import { ConfigureReplicaCallback, ReplicaManager } from './ReplicaManager'
+import { ConfigureReplicaCallback, ReplicaManager, type ReplicaManagerOptions } from './ReplicaManager'
 
-export type ReplicasOptions = {
-  url: string | string[]
-}
+export type ReplicasOptions =
+  | {
+      url: string | string[]
+    }
+  | { replicas: PrismaClient[] }
```

Comment:
I changed the type as shown above. However, because TypeScript uses structural subtyping, specifying both `url` and `replicas` arguments at the same time does not result in a type error.

Reply:
I've changed the type as follows:

https://github.com/prisma/extension-read-replicas/pull/30/commits/c8cbeea445737d41494e2449e35ca264dfa7aaf1

A type error will occur if the `url` and `replicas` options are specified simultaneously.